tests/test_upload.py

```
@@ -9,32 +9,59 @@
 """
 
 import logging
-from typing import Generator, Tuple
+from typing import Generator, Tuple, Type
```

Ruff (0.15.15)

[warning] Import from `collections.abc` instead: `Generator`

Import from `collections.abc`

(UP035)